modules/bullet/bullet_physics_server.cpp

```diff
@@ -49,78 +49,85 @@
 	@author AndreaCatania
 */
 
-#define CreateThenReturnRID(owner, ridData) \
-	RID rid = owner.make_rid(ridData);      \
-	ridData->set_self(rid);                 \
-	ridData->_set_physics_server(this);     \
+#define CreateRID(owner, ridData)      \
+	RID rid = owner.make_rid(ridData); \
+	ridData->set_self(rid);            \
+	ridData->_set_physics_server(this);
+
+#define CreateRIDReturn(owner, ridData) \
+	RID rid = owner.make_rid(ridData);  \
+	ridData->set_self(rid);             \
+	ridData->_set_physics_server(this); \
 	return rid;
 
-// <--------------- Joint creation asserts
-/// Assert the body is assigned to a space
-#define JointAssertSpace(body, bIndex, ret)                                                          \
+#define JointAssertSpace(body, bIndex)                                                               \
 	if (!body->get_space()) {                                                                        \
 		ERR_PRINT("Before create a joint the Body" + String(bIndex) + " must be added to a space!"); \
-		return ret;                                                                                  \
+		return;                                                                                      \
 	}
 
 /// Assert the two bodies of joint are in the same space
-#define JointAssertSameSpace(bodyA, bodyB, ret)                                                   \
+#define JointAssertSameSpace(bodyA, bodyB)                                                        \
 	if (bodyA->get_space() != bodyB->get_space()) {                                               \
 		ERR_PRINT("In order to create a joint the Body_A and Body_B must be in the same space!"); \
-		return RID();                                                                             \
+		return;                                                                                   \
 	}
 
 #define AddJointToSpace(body, joint) \
 	body->get_space()->add_constraint(joint, joint->is_disabled_collisions_between_bodies());
+
 // <--------------- Joint creation asserts
 
 void BulletPhysicsServer3D::_bind_methods() {
 	//ClassDB::bind_method(D_METHOD("DoTest"), &BulletPhysicsServer3D::DoTest);
 }
 
-BulletPhysicsServer3D::BulletPhysicsServer3D() :
-		PhysicsServer3D() {}
+BulletPhysicsServer3D::BulletPhysicsServer3D(bool p_using_threads) {
+	using_threads = p_using_threads;
+	active = true;
+	doing_sync = false;
```

Comment:
`active` and `doing_sync` are already initialized in the header, they don't need to be set here.